pymc/aesaraf.py

```diff
@@ -535,55 +535,155 @@ def make_shared_replacements(point, vars, model):
 
 def join_nonshared_inputs(
     point: Dict[str, np.ndarray],
-    xs: List[TensorVariable],
-    vars: List[TensorVariable],
-    shared,
-    make_shared: bool = False,
-):
+    outputs: List[TensorVariable],
+    inputs: List[TensorVariable],
+    shared_inputs: Optional[Dict[TensorVariable, TensorSharedVariable]] = None,
+    make_inputs_shared: bool = False,
+) -> Tuple[List[TensorVariable], TensorVariable]:
     """
-    Takes a list of Aesara Variables and joins their non shared inputs into a single input.
+    Create new outputs and input TensorVariables where the non-shared inputs are joined
+    in a single raveled vector input.
 
     Parameters
     ----------
-    point: a sample point
-    xs: list of Aesara tensors
-    vars: list of variables to join
+    point : Point
```

Comment:
I am not sure point is the right type. It does not match the type signature and it is not a "metatype" meaning a subset of dicts that match these properties but a different amd specific class. Does anybody know?

Reply:
Yeah. I'm not sure either. I was going off `Point` being in [reference link](https://pymc-data-umbrella.xyz/en/latest/sprint/tutorials/docstring_tutorial.html#parameters) and the fact that `Point` function would [construct the same type](https://pymcio--6216.org.readthedocs.build/projects/docs/en/6216/_modules/pymc/model.html#Point) as the alternative. 
i.e. `Dict[str, np.ndarray]`

let me know which fits best with the package

Reply:
`Dict[str, np.ndarray]` is the correct type here, but I'd say `Point` is also fine because users/devs probably understand what is meant in this context.

With the exact type hints in the signature, I treat the hints in the docstring as the "human readable" version, and often times I leave them away to not duplicate the information.

What is our policy, @OriolAbril ?

Reply:
Then this is what should be used (with the correft numpydoc formatting) instead of point which is therefore an invalid type. This function is on the website and it needs to have the type info properly documented in the docstring having users in mind

Reply:
Okay. So I am hearing to switch back to the `dict of {str : array_like}`. Is that correct?

Reply:
Why is Point referenced in the docs/.../conf.py
Is there a purpose or usecase?